mestopy/mestopy.py

```diff
@@ -1 +1,242 @@
+        return repr_string
+
+
+# Class for MeasurementChain as frame for RefObjs and calibration
+class MeasurementChain(object):
+    """Class for complete measurement chain.
+
+    This class that holds methods and properties of all devices in the
+    measurement chain. It can include a single or multiple devices from
+    'DeviceObj' class.
+    """
+
+    def __init__(self,
+                 sampling_rate,
+                 sound_device=None,
+                 devices=[],
+                 comment=None):
+        """Init measurement chain with rampling rate.
+
+        Attributes
+        ----------
+        sampling_rate : double
+            Sampling rate in Hertz.
+        sound_device : int
+            Number to identify the sound device used. The default is None.
+        devices : list
+            A list of instances from 'DeviceObj' class. The default is an
+            empty list.
+        comment : str
+            A comment related to the measurement chain. The default is None.
+        """
+        self.sampling_rate = sampling_rate
+        self.sound_device = sound_device
+        self.devices = devices
+        self.comment = comment
+
+    def add_device(self,
+                   device_data=None,
+                   sens=1,
+                   device_name=''):
+        """Adds a new device to the measurement chain.
+
+        Attributes
+        ----------
+        device_data : Signal, optional
+            Signal data that reprensets the inverted frequency response of the
+            device to add. The default is None, in this case a perfect flat
+            frequency response is assumed and only sensitivity as
+            a factor is applied.
+            Caution: Make sure to use frequency responses without artefacts,
+            as they will influence the quality of compensations calculated with
+            the measurement chain. Most ideal this should be regularized
+            inverted frequency responses. Data can be in domain 'freq' or
+            'time' and will be transformed if necessary.
+        sens : float, optional
+            Sensitivity of the device as a factor. If neither device_data nor
+            sens is given, add_device generates a device that has no effect to
+            the measurement chain as it has no frequency response and a
+            sesitivity (factor) of 1.
+        device_name : str, optional
+            The name of the new device to add to the masurement chain.
+            The default is an empty string.
+        """
+        if device_data is None:
+            device_data = Signal(np.full(int(self.sampling_rate/2), 1.0),
+                                 self.sampling_rate,
+                                 domain='freq')
+        # check if ref_signal is a pyfar.Signal, if not raise Error
+        if not isinstance(device_data, Signal):
+            raise TypeError('Input data must be of type: Signal.')
+        # check if there are no devices in measurement chain
```

Comment:
- I would move this to the Device Class.
- You don't have to generate a signal if `data==None` only using the sensitivity would work as well